src/ast/ast_annotations_ctx_collection.h

```diff
@@ -0,0 +1,41 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#pragma once
+#include "cypher-parser.h"
+
+typedef cypher_ast_annotation_context_t AnnotationCtx;
+
+typedef struct {
+	AnnotationCtx *name_ctx;        // Annotation context for naming graph entities and ORDER items.
+	AnnotationCtx *params_ctx;      // Annotation context for query parameters.
+	AnnotationCtx *project_all_ctx; // Context containing aliases for WITH/RETURN * projections.
+	AnnotationCtx *named_paths_ctx; // Annotation context for named paths projections.
+} AST_AnnotationCtxCollection;
+
+AST_AnnotationCtxCollection *AST_AnnotationCtxCollection_New();
+
+AnnotationCtx *AST_AnnotationCtxCollection_GetNameCtx(AST_AnnotationCtxCollection
```

Comment:
All getters should be operating on a `const AST_AnnotationCtxCollection *`

Reply:
done